core/src/main/java/org/mule/runtime/core/internal/routing/UntilSuccessfulRouter.java

```diff
@@ -273,15 +274,19 @@ private CoreEvent eventWithCurrentContextDeleted(CoreEvent event) {
 
   private Function<Throwable, Throwable> getThrowableFunction(CoreEvent event) {
     return throwable -> {
-      Throwable cause = getMessagingExceptionCause(throwable);
       CoreEvent exceptionEvent = event;
+      Throwable retryPolicyExhaustionCause = getMessagingExceptionCause(throwable);
+      // ConnectionException is treated in a way that prioritize it's error type over any other (see ErrorTypeLocator#getErrorTypeFromException)
+      retryPolicyExhaustionCause =
+          suppressIfPresent(retryPolicyExhaustionCause, ConnectionException.class, false);
```

Comment:
the cause should be logged in this case to help in troubleshooting.

Reply:
I will refactor this section in order to use the same approach (SuppressedException + Caused By)